node/src/block_ranges.rs

```diff
@@ -443,6 +457,33 @@ impl Sub<&BlockRanges> for BlockRanges {
     }
 }
 
+impl Sub<RangeInclusive<u64>> for BlockRanges {
+    type Output = Self;
+
+    fn sub(mut self, range: RangeInclusive<u64>) -> Self::Output {
+        if !range.is_empty() {
+            self.remove_relaxed(range).expect("shouldn't panic");
+        }
+        self
+    }
+}
+
+impl SubAssign<RangeInclusive<u64>> for BlockRanges {
+    fn sub_assign(&mut self, range: RangeInclusive<u64>) {
+        if !range.is_empty() {
+            self.remove_relaxed(range).expect("shouldn't panic");
+        }
+    }
+}
```

Comment:
I think I don't like them. IMO, if we do something like this, we should make `remove_relaxed` infallible, which opens other questions in that case.

Reply:
We can just return in `insert_relaxed` and `remove_relaxed` if range does not pass `validate()`. However I'm not sure if this will enable us to do any mistakes.

@zvolin Wdyt?

Reply:
My mental model for ranges is that of algebra of sets, so even though subtrahend here could be outside of `HeaderRange` strict domain, there's still a single, well-defined, and intuitive product we can return instead of erroring.
Since headers are going to be removed one by one, using `pop_tail`, I left these `impl`s mostly out of sentiment, so we can consider adding operators separately, if that's controversial.

Reply:
In case of sets, that means `10..=5` should be equal to `5..=10`. Which again it may allow us to do mistakes.